awx/main/dispatch/worker/callback.py

```diff
@@ -191,7 +192,7 @@ def flush(self, force=False):
                             e._retry_count = retry_count
 
                             # special sanitization logic for postgres treatment of NUL 0x00 char
-                            if (retry_count == 1) and isinstance(exc_indv, ValueError) and ("\x00" in e.stdout):
+                            if (retry_count == 1) and isinstance(exc_indv, DataError) and ("\x00" in e.stdout):
```

Comment:
Looks like this is the cause of

```
[gw0] linux -- Python 3.9.16 /var/lib/awx/venv/awx/bin/python3.9
Traceback (most recent call last):
  File "/awx_devel/awx/main/tests/functional/commands/test_callback_receiver.py", line 140, in test_postgres_invalid_NUL_char
    assert "\x00" not in events[0].stdout
AssertionError: assert '\x00' not in '\x00'
  '\x00' is contained here:
```

failure, which should be trivial to resolve

Reply:
Yea, it seems like two sources are seeing different raises. When I change between `ValueError` adn `DataError` two different tests fail depending on which is in here. So I simplified the if condition to remove the `isinstance` check and just wipe out the `\x00` if its there to handle both errors. Everything is green now so hopefully we can move forward.

Reply:
looks fine, this sort of thing seems expected.